fix(run): add ctx to run.Store

Arqu opened the pull request:
Also added an `owner profile.ID` to `ListByStatus` as cloud will have to scope it down and from a multi tenancy standpoint makes sense to control for it. And I want to avoid hacks we used to do where we magically pull out the relevant user from context.

b5 commented on automation/run/store.go:
```diff
@@ -25,26 +27,26 @@ var (
+	ListByStatus(ctx context.Context, owner profile.ID, s Status, lp params.List) ([]*State, error)
 	// Shutdown closes the store
-	Shutdown() error
+	Shutdown(ctx context.Context) error
```
wait, do we _really_ need a context on shutdown? Strikes me as a bad idea

Arqu replied:
Probably not, cloud should have a separate shutdown mechanic anyways (and does).

b5 reviewed (commented):
I'm not _so_ into adding contexts everywhere, but see how this is an important stop-gap in the short term. If we can nix it from the shutdown method, looks good

Arqu commented:
Updated the PR with a bit more stuff, the `step.Duration` can exceed a 32bit int and was giving me some trouble, so aligned that and also removed `ctx` from the `AddEvent` handler as that needs to run on a separate context from the event caller.
